extensions/amp-story/1.0/amp-story-page-attachment.js

```diff
@@ -133,25 +195,61 @@ export class AmpStoryPageAttachment extends DraggableDrawer {
    * @override
    */
   open(shouldAnimate = true) {
+    if (this.state_ === DrawerState.OPEN) {
+      return;
+    }
+
     super.open(shouldAnimate);
 
     this.storeService_.dispatch(Action.TOGGLE_SYSTEM_UI_IS_VISIBLE, false);
 
-    const currentHistoryState = /** @type {!Object} */ (getState(
-      this.win.history
-    ));
-    const historyState = {
-      ...currentHistoryState,
-      [HistoryState.ATTACHMENT_PAGE_ID]: this.storeService_.get(
-        StateProperty.CURRENT_PAGE_ID
-      ),
-    };
-
-    this.historyService_.push(() => this.closeInternal_(), historyState);
+    if (this.type_ === AttachmentType.INLINE) {
+      const currentHistoryState = /** @type {!Object} */ (getState(
+        this.win.history
+      ));
+      const historyState = {
+        ...currentHistoryState,
+        [HistoryState.ATTACHMENT_PAGE_ID]: this.storeService_.get(
+          StateProperty.CURRENT_PAGE_ID
+        ),
+      };
+      this.historyService_.push(() => this.closeInternal_(), historyState);
+    }
+
     this.analyticsService_.triggerEvent(StoryAnalyticsEvent.OPEN, this.element);
     this.analyticsService_.triggerEvent(
       StoryAnalyticsEvent.PAGE_ATTACHMENT_ENTER
     );
+
+    if (this.type_ === AttachmentType.REMOTE) {
+      this.openRemote_();
+    }
+  }
+
+  /**
+   * Triggers a remote attachment opening animation, and redirects to the
+   * specified URL.
+   * @private
+   */
+  openRemote_() {
+    const animationEl = this.win.document.createElement('div');
+    animationEl.classList.add('i-amphtml-story-page-attachment-expand');
+    const storyEl = closest(this.element, el => el.tagName === 'AMP-STORY');
+    this.storeService_.dispatch(Action.TOGGLE_SYSTEM_UI_IS_VISIBLE, false);
+
+    this.mutateElement(() => {
+      storyEl.appendChild(animationEl);
+    }).then(() => {
+      // Give some time for the 120ms animation to run. The navigation itself
```

Comment:
nit: can you add to the comment to describe where that 120ms is defined?

Reply:
Done